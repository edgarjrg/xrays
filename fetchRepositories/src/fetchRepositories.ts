import { lensProp, lensPath, transduce, view, compose, map } from "ramda";

export class Fetcher {

}

export const valuesLens = lensProp('values')
export const linkLens = lensPath(['links', 'self', 'href'])

export function getLinks(input) {
  return map(view(linkLens), view(valuesLens, input))
}