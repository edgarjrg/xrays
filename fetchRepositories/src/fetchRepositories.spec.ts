import { valuesLens, linkLens, getLinks } from './fetchRepositories'
import { view } from 'ramda'

test('valuesLens should bring the value of response', async () => {
  const PRsReponse = require('./testData/mergedPRsResponse.json')
  const valuesLensResponse = require('./testData/valuesLensResponse.json')
  expect(view(valuesLens, PRsReponse)).toEqual(valuesLensResponse)
});

test('linkLens should bring the link of the PR', async () => {
  const linkLensInput = require('./testData/linkLens.input.json')
  const linkLensExpected = require('./testData/linkLens.expected.json')
  expect(view(linkLens, linkLensInput)).toEqual(linkLensExpected)
});

test('getLinks should bring the link of the PR', async () => {
  const PRsReponse = require('./testData/mergedPRsResponse.json')
  const linkLensExpected = require('./testData/linkLens.expected.json')
  expect(getLinks(PRsReponse)).toMatchSnapshot()
});

