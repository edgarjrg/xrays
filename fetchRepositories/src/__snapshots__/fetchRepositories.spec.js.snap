// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLinks should bring the link of the PR 1`] = `
Array [
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/35",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/34",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/33",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/32",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/31",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/30",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/29",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/28",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/26",
  "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/25",
]
`;
